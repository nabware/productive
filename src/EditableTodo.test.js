import React from "react";
import { render, fireEvent } from "@testing-library/react";
import EditableTodo from "./EditableTodo.js";
import TEST_TODOS from "./_testCommon.js";

describe("EditableTodo", function () {
  it("renders without crashing", function () {
    render(<EditableTodo todo={TEST_TODOS[0]} />);
  });

  it("displays todo, edit, and delete buttons", function () {
    const { getByText } = render(<EditableTodo todo={TEST_TODOS[0]} />);

    expect(getByText("test1title")).toBeInTheDocument();
    expect(getByText("Del")).toBeInTheDocument();
    expect(getByText("Edit")).toBeInTheDocument();
  });

  it("edit button shows form on click and hides on save", function () {
    const dumbyHandleSave = jest.fn();
    const { container, getByText } = render(<EditableTodo todo={TEST_TODOS[0]} update={dumbyHandleSave} />);

    expect(container.querySelector("#newTodo-title")).not.toBeInTheDocument();
    expect(container.querySelector("#newTodo-description")).not.toBeInTheDocument();
    expect(container.querySelector("#newTodo-priority")).not.toBeInTheDocument();

    const editButton = getByText("Edit");
    fireEvent.click(editButton);

    expect(container.querySelector("#newTodo-title")).toBeInTheDocument();
    expect(container.querySelector("#newTodo-description")).toBeInTheDocument();
    expect(container.querySelector("#newTodo-priority")).toBeInTheDocument();

    const saveButton = getByText("GÃ¸!");
    fireEvent.click(saveButton);

    expect(container.querySelector("#newTodo-title")).not.toBeInTheDocument();
    expect(container.querySelector("#newTodo-description")).not.toBeInTheDocument();
    expect(container.querySelector("#newTodo-priority")).not.toBeInTheDocument();
    expect(dumbyHandleSave).toHaveBeenCalledTimes(1);
  });

  it("delete button calls delete function", function () {
    const dumbyHandleDel = jest.fn();
    const { getByText } = render(<EditableTodo todo={TEST_TODOS[0]} remove={dumbyHandleDel} />);

    const deleteButton = getByText("Del");
    fireEvent.click(deleteButton);

    expect(dumbyHandleDel).toHaveBeenCalledTimes(1);
  });

  it("snapshot", function () {
    const { container } = render(<EditableTodo todo={TEST_TODOS[0]} />);
    expect(container).toMatchSnapshot();
  });
});

