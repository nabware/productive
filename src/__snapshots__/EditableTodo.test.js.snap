// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditableTodo snapshot 1`] = `
<div>
  <div
    class="EditableTodo"
  >
    <div
      class="mb-3"
    >
      <div
        class="float-end text-sm-end"
      >
        <button
          class="EditableTodo-toggle btn-link btn btn-sm"
        >
          Edit
        </button>
        <button
          class="EditableTodo-delBtn btn-link btn btn-sm text-danger"
        >
          Del
        </button>
      </div>
      <div
        class="Todo"
      >
        <div>
          <b>
            test1title
          </b>
           
          <small>
            (priority: 
            1
            )
          </small>
        </div>
        <div>
          <small>
            test1description
          </small>
        </div>
      </div>
    </div>
  </div>
</div>
`;
